chai/chai-tests.ts

```diff
@@ -5,290 +5,416 @@ import chai = require('chai');
+    (0).should.not.be.ok;
 
     err(() => {
         expect('').to.be.ok;
+        ''.should.be.ok;
     }, 'expected \'\' to be truthy');
 
     err(() => {
         expect('test').to.not.be.ok;
+        'test'.should.not.be.ok;
     }, 'expected \'test\' to be falsy');
 }
 
 function _false() {
     expect(false).to.be.false;
+    false.should.be.false;
     expect(true).to.not.be.false;
+    true.should.not.be.false;
     expect(0).to.not.be.false;
+    (0).should.not.be.false;
 
     err(() => {
         expect('').to.be.false;
+        ''.should.be.false;
     }, 'expected \'\' to be false');
 }
 
 function _null() {
     expect(null).to.be.null;
+    should.equal(null, null);
     expect(false).to.not.be.null;
+    false.should.not.be.null;
 
     err(() => {
         expect('').to.be.null;
+        ''.should.be.null;
     }, 'expected \'\' to be null');
 }
 
 function _undefined() {
     expect(undefined).to.be.undefined;
+    should.equal(undefined, undefined);
     expect(null).to.not.be.undefined;
+    should.not.equal(null, undefined);
 
     err(() => {
         expect('').to.be.undefined;
+        ''.should.be.undefined;
     }, 'expected \'\' to be undefined');
 }
 
 function exist() {
     var foo = 'bar';
     expect(foo).to.exist;
+    should.exist(foo);
     expect(void(0)).to.not.exist;
+    should.not.exist(void (0));
 }
 
 function arguments() {
     var args = arguments;
     expect(args).to.be.arguments;
+    args.should.be.arguments;
     expect([]).to.not.be.arguments;
+    [].should.not.be.arguments;
     expect(args).to.be.an('arguments').and.be.arguments;
+    args.should.be.an('arguments').and.be.arguments;
     expect([]).to.be.an('array').and.not.be.Arguments;
+    [].should.be.an('array').and.not.be.Arguments;
 }
 
 function equal() {
     expect(undefined).to.equal(void(0));
+    should.equal(undefined, void(0));
 }
 
 function _typeof() {
     expect('test').to.be.a('string');
+    'test'.should.be.a('string');
 
     err(() => {
         expect('test').to.not.be.a('string');
+        'test'.should.not.be.a('string');
     }, 'expected \'test\' not to be a string');
 
     expect(arguments).to.be.an('arguments');
+    arguments.should.be.an('arguments');
 
     expect(5).to.be.a('number');
+    (5).should.be.a('number');
+
     expect(new Number(1)).to.be.a('number');
+    (new Number(1)).should.be.a('number');
     expect(Number(1)).to.be.a('number');
+    Number(1).should.be.a('number');
     expect(true).to.be.a('boolean');
+    true.should.be.a('boolean');
     expect(new Array()).to.be.a('array');
+    (new Array()).should.be.a('array');
     expect(new Object()).to.be.a('object');
+    (new Object()).should.be.a('object');
     expect({}).to.be.a('object');
+    ({}).should.be.a('object');
     expect([]).to.be.a('array');
+    [].should.be.a('array');
     expect(() => { }).to.be.a('function');
+    (() => { }).should.be.a('function');
     expect(null).to.be.a('null');
+    // has no should equivalent?
```

Comment:
That would make sense. See MDN for the [difference between null and undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null#Difference_between_null_and_undefined):
```js
typeof null        // object (bug in ECMAScript, should be null)
typeof undefined   // undefined
null === undefined // false
null  == undefined // true
```
Also, you can run this in your console:
```js
(null).prototype // Uncaught: TypeError: Cannot read property 'prototype' of null
```